internal/witness/witness_test.go

```diff
@@ -324,6 +325,88 @@ func TestWitness_UpdateResponse(t *testing.T) {
 	}
 }
 
+func TestWitnessStateEvolution(t *testing.T) {
+	logSigner := mustCreateSigner(t, log_skey)
+
+	logSignedCheckpoint, err := note.Sign(&note.Note{Text: cp}, logSigner)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Set up a fake server hosting the witnesses.
+	// The witnesses just sign the checkpoint with whatever key is requested, they don't check the body at all.
+	// An improvement on this would be to make the fake witnesses more realistic, but it's a non-trivial
+	// amount of code to add to this already long test!
+	var wit1 tessera.Witness
+	var count int
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w1u := mustUrl(t, wit1.Url)
+		if got, want := r.URL.String(), w1u.Path; got != want {
+			t.Fatalf("got request to URL %q but expected %q", got, want)
+		}
+
+		switch count {
+		case 0:
+			w.Header().Add("Content-Type", "text/x.tlog.size")
+			w.WriteHeader(409)
+			_, _ = w.Write([]byte("1000"))
+		case 1:
+			body, err := io.ReadAll(r.Body)
+			if err != nil {
+				t.Fatal(err)
+			}
+			if !bytes.HasPrefix(body, []byte("old 1000")) {
+				t.Fatalf("expected body to start with old 1000 but got\n%v", body)
+			}
+
+			_, _ = w.Write(sigForSigner(t, cp, wit1_skey))
+		case 2:
+			body, err := io.ReadAll(r.Body)
+			if err != nil {
+				t.Fatal(err)
+			}
+			if !bytes.HasPrefix(body, []byte("old 34840403")) {
+				t.Fatalf("expected body to start with old 34840403 but got\n%v", string(body))
+			}
```

Comment:
Do we hit here? I can see two calls to `g.Witness` below which I imagine would hit `count==0` and then `count==`?